mantis-control-plane/mantis-control-plane-core/src/main/java/io/mantisrx/server/core/scheduler/SchedulingConstraints.java

```diff
@@ -20,23 +20,36 @@
 import io.mantisrx.shaded.com.google.common.annotations.VisibleForTesting;
 import io.mantisrx.shaded.com.google.common.collect.ImmutableMap;
 import java.util.Map;
+import java.util.Optional;
 import lombok.AllArgsConstructor;
 import lombok.Value;
 
 /**
- * A class that represents scheduling constraints. These constraints include the resource constraints, and a map of scheduling attributes (e.g. jdkVersion:17).
+ * A class that represents scheduling constraints.
+ * These constraints include the resource constraints for scheduling,
+ * the size name that could match the size label of a Task Executor,
+ * and a map of scheduling attributes (e.g., jdkVersion:17).
  */
 @AllArgsConstructor(staticName = "of")
 @Value
 public class SchedulingConstraints {
     // Defines the resource constraints for scheduling
     MachineDefinition machineDefinition;
 
+    // Optional field to set predefined size name. When present, the scheduling system tries
+    // to match this field with the size name of a Task Executor.
+    Optional<String> sizeName;
```

Comment:
Yeah the idea is to be able to support both scheduling with size name & machine definition but for now I'm assuming size name to be optional while machine definition mandatory. this might change in the future though.

Reply:
I improved the docs. let me know if that's still not enough.